### DISCUSSION TITLE
adding google AdMob

### DISCUSSION THREAD

**Discussion by kebele** [quasarframework/quasar#9638] [CLI - Capacitor / Cordova modes] [answered]
I made an android application with Quasar and published it on the play store. 
Now I want to add google admob to my app. 
How can I do this? 
Is there a guide or similar document? 
Thank you

**Comment by ontwikkelfabriek** [comment, marked as answer]
@kebele 

Did you create the app with Capacitor or Cordova?

For Cordova I would start here:
https://admob-plus.github.io/docs/cordova

For Capacitor I would start here:
https://admob-plus.github.io/docs/capacitor

**Reply by kebele** [reply]
I followed the instructions suggested by Quasar framework.
According to it, the application is created with cordova. 
I will try the solution you shared as soon as possible then will share the result here.
Thank you.

**Comment by kebele** [comment]
I solved my problem with a little effort and the plugin you shared. 
Thank you.

**Reply by OziOcb** [reply]
Hi. Could you share a link to the GitHub project where you implemented that, please?

**Reply by capslockoff** [reply]
@kebele  I am running into some issues getting the admob-plus to work with cordova. I am running into some kind of dependency issue.

Do you mind sharing your `android/app/build.gradle` dependencies?

Mine is like this and I have a feeling something is a mismatch

```
    implementation "com.google.android.gms:play-services-auth:15.0.1"
    implementation "com.google.android.gms:play-services-identity:15.0.1"
    implementation "com.google.android.gms:play-services-ads:20.4.0"
```

```
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:
  Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
  Program type already present: com.google.android.gms.common.api.internal.zzc
```

**Reply by kebele** [reply]
implementation "com.google.android.gms:play-services-ads:20.1.0"

> @kebele I am running into some issues getting the admob-plus to work with cordova. I am running into some kind of dependency issue.
> 
> Do you mind sharing your `android/app/build.gradle` dependencies?
> 
> Mine is like this and I have a feeling something is a mismatch
> 
> ```
>     implementation "com.google.android.gms:play-services-auth:15.0.1"
>     implementation "com.google.android.gms:play-services-identity:15.0.1"
>     implementation "com.google.android.gms:play-services-ads:20.4.0"
> ```
> 
> ```
> Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:
>   Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
>   Program type already present: com.google.android.gms.common.api.internal.zzc
> ```

Hi, there is no dependency information about ...services-auth and ...services-identity in my files. Among the information you want, there is only this:
implementation "com.google.android.gms:play-services-ads:20.1.0"

**Reply by capslockoff** [reply]
Adding this here for anyone else that is using `google-plus` login along with `admob-plus`. You will need to set the `PLAY_SERVICES_VERSION` for both of the plugins to `16.0.0` That is the only combination that worked for me.

**Comment by kebele** [comment]
> Hi. Could you share a link to the GitHub project where you implemented that, please?

Sorry for the late reply, not a project that I shared on Github

**Comment by ASQUAREEE** [comment]
@kebele  bro i am still struggling with admob plus , how to import cordova plugin into vue component , please help me, I will give u a treat for sure

**Comment by kebele** [comment]
It has been a long time since I did this, so I don't remember btw I'm on a vacation. I will check when I get home. As fas as I remember I folloed the official document.

Android için Outlook<https://aka.ms/AAb9ysg> edinin
________________________________

**Reply by ASQUAREEE** [reply]
thank you for responding , "No worries! Enjoy your vacation! When you get back home,  then please check it once

**Comment by EMI-INDO** [comment]
You can also try this https://github.com/quasarframework/quasar/discussions/17706
